web/app/js/components/Navigation.jsx

```diff
@@ -245,6 +288,14 @@ class NavigationBase extends React.Component {
               <ListItemIcon><LibraryBooksIcon /></ListItemIcon>
               <ListItemText primary="Documentation" />
             </ListItem>
+            { this.menuItem("/community", "Community",
+              <Badge
+                classes={{ badge: classes.badge }}
+                invisible={this.state.hideUpdateBadge}
+                badgeContent="1">
```

Comment:
It seems like we could set the number in the badge to be the actual number of updates since the user last clicked, based on the timestamp of each of the updates in the JSON feed. How would you feel about something like?

```diff
diff --git a/web/app/js/components/Navigation.jsx b/web/app/js/components/Navigation.jsx
index 2434bd03..365ef498 100644
--- a/web/app/js/components/Navigation.jsx
+++ b/web/app/js/components/Navigation.jsx
@@ -27,7 +27,7 @@ import SentimentVerySatisfiedIcon from '@material-ui/icons/SentimentVerySatisfie
 import Toolbar from '@material-ui/core/Toolbar';
 import Typography from '@material-ui/core/Typography';
 import Version from './Version.jsx';
-import _maxBy from 'lodash/maxBy';
+import _filter from 'lodash/filter';
 import classNames from 'classnames';
 import { withContext } from './util/AppContext.jsx';
 import { withStyles } from '@material-ui/core/styles';
@@ -149,7 +149,7 @@ class NavigationBase extends React.Component {
     return {
       drawerOpen: true,
       helpMenuOpen: false,
-      hideUpdateBadge: true,
+      communityUpdateCount: 0,
       latestVersion: '',
       isLatest: true,
       namespaceFilter: "all"
@@ -186,14 +186,13 @@ class NavigationBase extends React.Component {
         if (rsp.length > 0) {
           let lastClicked = localStorage["linkerd-updates-last-clicked"];
           if (!lastClicked) {
-            this.setState({ hideUpdateBadge: false });
+            this.setState({ communityUpdateCount: rsp.length });
           } else {
             let lastClickedDateObject = new Date(lastClicked);
-            let latestArticle = _maxBy(rsp, update => update.date);
-            let latestArticleDateObject = new Date(latestArticle);
-            if (latestArticleDateObject > lastClickedDateObject) {
-              this.setState({ hideUpdateBadge: false });
-            }
+            let newArticles = _filter(rsp, update => {
+              return new Date(update.date) > lastClickedDateObject;
+            });
+            this.setState({ communityUpdateCount: newArticles.length });
           }
         }
       }).catch(this.handleApiError);
@@ -208,9 +207,7 @@ class NavigationBase extends React.Component {
   handleCommunityClick = () => {
     let lastClicked = new Date();
     localStorage.setItem("linkerd-updates-last-clicked", lastClicked);
-    if (this.state.hideUpdateBadge === false) {
-      this.setState({ hideUpdateBadge: true });
-    }
+    this.setState({ communityUpdateCount: 0 });
   }
 
   handleDrawerClick = () => {
@@ -291,8 +288,8 @@ class NavigationBase extends React.Component {
             { this.menuItem("/community", "Community",
               <Badge
                 classes={{ badge: classes.badge }}
-                invisible={this.state.hideUpdateBadge}
-                badgeContent="1">
+                invisible={this.state.communityUpdateCount === 0}
+                badgeContent={this.state.communityUpdateCount}>
                 <SentimentVerySatisfiedIcon />
               </Badge>, this.handleCommunityClick
               ) }
```

Reply:
Thanks for these suggestions, and I will save this to look at in the future when we improve this feature. Initially we are hard-coding the number of updates to "1" in order to not overwhelm the users, but this may change!